File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/java/series2d/HitTestAPI.java
package com.scichart.docsandbox.examples.java.series2d;

import android.graphics.PointF;
import android.view.MotionEvent;
import android.view.View;

import androidx.annotation.NonNull;

import com.scichart.charting.visuals.SciChartSurface;
import com.scichart.charting.visuals.renderableSeries.IRenderableSeries;
import com.scichart.charting.visuals.renderableSeries.hitTest.HitTestInfo;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.examples.base.SingleChart2DFragment;

@ExampleDefinition()
public class HitTestAPI extends SingleChart2DFragment {
    private final PointF touchPoint = new PointF();
    private final HitTestInfo hitTestInfo = new HitTestInfo();

    @Override
    protected void initExample(@NonNull SciChartSurface surface) { }

    // <PerformHitTest>
    boolean onTouch(View v, MotionEvent event) {
        final SciChartSurface surface = (SciChartSurface) v;

        // The touch point relative to the ChartSurface
        touchPoint.set(event.getX(), event.getY());
        // Translate the touch point relative to RenderableSeriesArea (or ModifierSurface)
        surface.translatePoint(touchPoint, surface.getRenderableSeriesArea());

        for (IRenderableSeries renderableSeries : surface.getRenderableSeries()) {
            // Perform `Hit-Test` which will be stored in the `_hitTestInfo`
            renderableSeries.hitTest(hitTestInfo, touchPoint.x, touchPoint.y);
        }

        return true;
    }
    // </PerformHitTest>
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/series2d/HitTestAPI.kt
package com.scichart.docsandbox.examples.kotlin.series2d

import android.graphics.PointF
import android.view.MotionEvent
import android.view.View
import com.scichart.charting.visuals.SciChartSurface
import com.scichart.charting.visuals.renderableSeries.hitTest.HitTestInfo
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.examples.base.SingleChart2DFragment

@ExampleDefinition()
class HitTestAPI : SingleChart2DFragment() {
    private val touchPoint = PointF()
    private val hitTestInfo = HitTestInfo()
    
    override fun initExample(surface: SciChartSurface) {}

    // <PerformHitTest>
    fun onTouch(v: View, event: MotionEvent): Boolean {
        val surface = v as SciChartSurface

        // The touch point relative to the ChartSurface
        touchPoint.set(event.x, event.y)
        // Translate the touch point relative to RenderableSeriesArea (or ModifierSurface)
        surface.translatePoint(touchPoint, surface.renderableSeriesArea)
        for (renderableSeries in surface.renderableSeries) {
            // Perform `Hit-Test` which will be stored in the `_hitTestInfo`
            renderableSeries.hitTest(hitTestInfo, touchPoint.x, touchPoint.y)
        }
        return true
    }
    // </PerformHitTest>
}
